deploy/examples/create-external-cluster-resources.py

```diff
@@ -508,13 +504,10 @@ def get_active_and_standby_mgrs(self):
         mgr_ips.insert(0, monitoring_endpoint_ip)
         all_mgr_ips_str = ",".join(mgr_ips)
         return all_mgr_ips_str, monitoring_endpoint_port
-
-    def create_cephCSIKeyring_cephFSProvisioner(self):
-        '''
-        command: ceph auth get-or-create client.csi-cephfs-provisioner mon 'allow r' mgr 'allow rw' osd 'allow rw tag cephfs metadata=*'
-        '''
+    
+    def get_cephfs_provisioner_caps_and_entity(self):
```

Comment:
This function could be optimized,

```suggestion
    def get_cephfs_provisioner_caps_and_entity(self):
        entity = "client.csi-cephfs-provisioner"
        caps = {"mon": "allow r, allow command 'osd blocklist'",
                "mgr": "allow rw",
                "osd": "allow rw tag cephfs metadata=*"}
        if self._arg_parser.restricted_auth_permission:
            cluster_name = self._arg_parser.cluster_name
            if cluster_name == "":
                raise ExecutionFailureException(
                    "cluster_name not found, please set the '--cluster-name' flag")
            cephfs_filesystem = self._arg_parser.cephfs_filesystem_name
            if cephfs_filesystem == "":
                entity = "{}-{}".format(entity, cluster_name)
            else:
                entity = "{}-{}-{}".format(entity, cluster_name, cephfs_filesystem)
                caps["osd"] = "allow rw tag cephfs metadata={}".format(cephfs_filesystem)
        
        return caps,entity
```